internal/ingress/annotations/customheaders/main.go

```diff
@@ -0,0 +1,124 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package customheaders
+
+import (
+	"fmt"
+	"regexp"
+
+	"k8s.io/klog/v2"
+
+	networking "k8s.io/api/networking/v1"
+
+	"golang.org/x/exp/slices"
+	"k8s.io/ingress-nginx/internal/ingress/annotations/parser"
+	ing_errors "k8s.io/ingress-nginx/internal/ingress/errors"
+	"k8s.io/ingress-nginx/internal/ingress/resolver"
+)
+
+// Config returns the custom response headers for an Ingress rule
+type Config struct {
+	Headers map[string]string `json:"headers,omitempty"`
+}
+
+var (
+	headerRegexp = regexp.MustCompile(`^[a-zA-Z\d\-_]+$`)
+	valueRegexp  = regexp.MustCompile(`^[a-zA-Z\d_ :;.,\\/"'?!(){}\[\]@<>=\-+*#$&\x60|~^%]+$`)
```

Comment:
This value may be too wide. Can we limit it a bit? I would like to avoid the special characters (like (, {, ;, etc) and we expand it as needed. Is the 95% covered by alpha numeric, -, _, etc?

Reply:
If you ban characters like `;` then this feature won't be useful in many cases.
For example Content-Security-Policy header requires semicolons and other special characters. If we want to avoid surprises I think we should stick to the standard. You an find info here:
https://developers.cloudflare.com/rules/transform/response-header-modification/reference/header-format/

Reply:
i adopted the regex from the site @jacekn mentioned.

Reply:
@rikatz can we merge the pr now? or does something else need to be changed?